Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/RenamingPolicy.cs

```diff
@@ -1,270 +1,270 @@
+        var originalFileNameWithoutPath = Path.GetFileName(originalFileName);
 
-                return remoteFileDefinition;
-            }
+        if (string.IsNullOrEmpty(remoteFileDefinition))
+        {
+            return originalFileNameWithoutPath;
+        }
 
-            var result = this.ExpandMacrosAndMasks(originalFileName, remoteFileDefinition);
+        if (!IsFileMask(remoteFileDefinition) &&
+            !IsFileMacro(remoteFileDefinition, MacroHandlers) &&
+            !IsFileMacro(remoteFileDefinition, SourceFileNameMacroHandlers))
+        {
+            // remoteFileDefination does not have macros
+            var remoteFileName = Path.GetFileName(remoteFileDefinition);
 
-            if (result.EndsWith("\\"))
+            if (string.IsNullOrEmpty(remoteFileName))
             {
-                result = Path.Combine(result, originalFileNameWithoutPath);
+                remoteFileDefinition = Path.Combine(remoteFileDefinition, originalFileNameWithoutPath);
             }
 
-            return result;
+            return remoteFileDefinition;
         }
 
-        public string ExpandDirectoryForMacros(string directory)
-        {
-            if (directory.Contains("%SourceFileName%") || directory.Contains("%SourceFileExtension%"))
-                throw new Exception("'%SourceFileName%' and '%SourceFileExtension%' are not supported macros for source and destination directories.");
+        var result = this.ExpandMacrosAndMasks(originalFileName, remoteFileDefinition);
 
-            return ExpandFileMacros(directory);
-        }
-       
-        public string CreateRemoteFilePathForMove(string sourceOperationTo, string sourceFilePath)
+        if (result.EndsWith("\\"))
         {
-            var directoryName = sourceOperationTo;
-            if (string.IsNullOrEmpty(directoryName))
-            {
-                throw new ArgumentException("When using move as a source operation, you should always define a directory", "sourceOperationTo");
-            }
+            result = Path.Combine(result, originalFileNameWithoutPath);
+        }
 
+        return result;
+    }
 
-            directoryName = CanonizeAndCheckPath(directoryName);
+    public string ExpandDirectoryForMacros(string directory)
+    {
+        if (directory.Contains("%SourceFileName%") || directory.Contains("%SourceFileExtension%"))
+            throw new Exception("'%SourceFileName%' and '%SourceFileExtension%' are not supported macros for source and destination directories.");
 
-            // this should always be a directory
-            if (!directoryName.EndsWith("/"))
-            {
-                directoryName = directoryName + "/";
-            }
-            var sourceFileName = Path.GetFileName(sourceFilePath);
-            return Path.Combine(directoryName, sourceFileName);
+        return ExpandFileMacros(directory);
+    }
+       
+    public string CreateRemoteFilePathForMove(string sourceOperationTo, string sourceFilePath)
```

Comment:
This method can be static.

Reply:
If the method is static there's a conflict 
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0176	Member 'RenamingPolicy.CreateRemoteFilePathForMove(string, string)' cannot be accessed with an instance reference; qualify it with a type name instead	Frends.SFTP.UploadFiles (net471), Frends.SFTP.UploadFiles (net6.0), Frends.SFTP.UploadFiles (netstandard2.0)	C:\Users\virtari\repos\Frends.SFTP\Frends.SFTP.UploadFiles\Frends.SFTP.UploadFiles\Definitions\SingleFileTransfer.cs	263	Active
```